jmh-benchmarks/src/main/java/org/apache/kafka/jmh/assignor/TargetAssignmentBuilderBenchmark.java

```diff
@@ -119,62 +117,34 @@ public void setup() {
     }
 
-    private Map<String, ConsumerGroupMember> generateMockMembers() {
-        Map<String, ConsumerGroupMember> members = new HashMap<>();
+    private void setupTopics() {
+        allTopicNames = AssignorBenchmarkUtils.createTopicNames(topicCount);
 
-        for (int i = 0; i < memberCount - 1; i++) {
-            ConsumerGroupMember member = new ConsumerGroupMember.Builder("member" + i)
-                .setSubscribedTopicNames(allTopicNames)
-                .build();
-            members.put("member" + i, member);
-        }
-        return members;
-    }
+        int partitionsPerTopic = (memberCount * partitionsToMemberRatio) / topicCount;
+        subscriptionMetadata = AssignorBenchmarkUtils.createSubscriptionMetadata(
+            allTopicNames,
+            partitionsPerTopic,
+            topicName -> Collections.emptyMap()
+        );
 
-    private Map<String, TopicMetadata> generateMockSubscriptionMetadata() {
-        Map<String, TopicMetadata> subscriptionMetadata = new HashMap<>();
-        MetadataDelta delta = new MetadataDelta(MetadataImage.EMPTY);
-        int partitionsPerTopicCount = (memberCount * partitionsToMemberRatio) / topicCount;
-
-        for (int i = 0; i < topicCount; i++) {
-            String topicName = "topic-" + i;
-            Uuid topicId = Uuid.randomUuid();
-            allTopicNames.add(topicName);
-
-            TopicMetadata metadata = new TopicMetadata(
-                topicId,
-                topicName,
-                partitionsPerTopicCount,
-                Collections.emptyMap()
-            );
-            subscriptionMetadata.put(topicName, metadata);
-
-            AssignorBenchmarkUtils.addTopic(
-                delta,
-                topicId,
-                topicName,
-                partitionsPerTopicCount
-            );
-        }
+        topicsImage = AssignorBenchmarkUtils.createTopicsImage(subscriptionMetadata);
 
-        topicsImage = delta.apply(MetadataProvenance.EMPTY).topics();
-        return subscriptionMetadata;
+        Map<Uuid, TopicMetadata> topicMetadata = AssignorBenchmarkUtils.createTopicMetadata(subscriptionMetadata);
+        subscribedTopicDescriber = new SubscribedTopicDescriberImpl(topicMetadata);
     }
 
-    private Map<String, Assignment> generateMockInitialTargetAssignmentAndUpdateInvertedTargetAssignment() {
-        Map<Uuid, TopicMetadata> topicMetadataMap = new HashMap<>(topicCount);
-        subscriptionMetadata.forEach((topicName, topicMetadata) ->
-            topicMetadataMap.put(
-                topicMetadata.id(),
-                topicMetadata
-            )
+    private Map<String, Assignment> generateMockInitialTargetAssignmentAndUpdateInvertedTargetAssignment(
+        Map<String, ConsumerGroupMember> members
+    ) {
+        this.groupSpec = AssignorBenchmarkUtils.createGroupSpec(
+            members,
+            HOMOGENEOUS,
+            topicsImage
         );
 
-        createAssignmentSpec();
```

Comment:
`createAssignmentSpec()` populates `this.groupSpec`. This is now a call to `AssignorBenchmarkUtils.createGroupSpec()`.